R/get_source_expressions.R

```diff
@@ -502,7 +502,7 @@ get_source_expression <- function(source_expression, error = identity) {
     error = error
   )
 
-  if (inherits(parsed_content, c("error", "lint"))) {
+  if (is_error(parsed_content) || is_lint(parsed_content)) {
```

Comment:
This should be `&&` IINM.

Reply:
The original code is conveying that either of those classes is fine, or at least that's what `inherits()` entails:

``` r
x <- tibble::tibble()
y <- data.frame()

class(x)
#> [1] "tbl_df"     "tbl"        "data.frame"
class(y)
#> [1] "data.frame"

inherits(x, c("tbl_df", "data.frame"))
#> [1] TRUE
inherits(y, c("tbl_df", "data.frame"))
#> [1] TRUE
```

<sup>Created on 2024-11-03 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Reply:
Ah, thanks for clarifying.